### DISCUSSION TITLE
How to encrypt test passwords used in synthetic browser monitoring

### DISCUSSION THREAD

**Discussion by sirishav** [Dynatrace/dynatrace-configuration-as-code#751] [Q&A]
Hi,

We are using Monaco CLI to add synthetic browser monitoring and deploying them to new environments. One of the browser monitor is to login into an application with test username and test password. Dynatrace enters nothing in the password field. How to encrypt the test passwords that is acceptable by synthetic monitor.

**Comment by UnseenWizzard** [comment]
Hi @sirishav 
the general recommendation for this would be to use the [Crednetial Vault](https://www.dynatrace.com/support/help/how-to-use-dynatrace/synthetic-monitoring/general-information/credential-vault-for-synthetic-monitors) and reference these credentials from your synthetics.

You can create these credentials (type 'credential-vault') using monaco. 

You can find a bad example of this here: https://github.com/dynatrace-oss/dynatrace-monitoring-as-code/tree/main/cmd/monaco/test-resources/integration-all-configs/project/credential-vault

This sample sadly stores "credentials" in plaintext - the actually recommended pattern for this is to pass any credentials in via environment variables when running the deployment: 
https://dynatrace-oss.github.io/dynatrace-monitoring-as-code/configuration/yaml_config#templating-of-environment-variables
